cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -73,6 +76,18 @@ public static String getNamespacedBaseDir(NamespacedLocationFactory namespacedLo
     return namespacedLocationFactory.get(namespaceId.toId()).append(logBaseDir).toString();
   }
 
+  public static Location getNamespacedBaseDirLocation(final NamespacedLocationFactory namespacedLocationFactory,
```

Comment:
@bdmogal Its because the previous method is called from FileMetaDataManager#cleanMetaData() which already wraps the call with impersonation.